createdisk-library.sh

```diff
@@ -165,7 +165,7 @@ function prepare_hyperV() {
     else
         # Download the hyperV daemons and libvarlink-util dependency on host
         mkdir $1/packages
-        sudo yum install -y --downloadonly --downloaddir $1/packages hyperv-daemons
+        sudo yum download --downloadonly --downloaddir $1/packages hyperv-daemons --resolve
```

Comment:
This is part of `dnf-utils`, dunno if this gets installed by default? For what it's worth, `--resolve` will only pick up dependencies if they are not already installed on the system, so this has the same issue as what you are trying to solve, but only for dependencies (ie `hyperv-daemons` will always be downloaded).

Reply:
> This is part of dnf-utils, dunno if this gets installed by default?

It is by default installed.

>  For what it's worth, --resolve will only pick up dependencies if they are not already installed on the system, so this has the same issue as what you are trying to solve, but only for dependencies (ie hyperv-daemons will always be downloaded).

@cfergeau Ah yes looks like I need to use `alldep` also.

```
--alldeps
              When used with --resolve, download all dependencies (do not skip already installed ones).
```

Reply:
Looks like `alldep` is not working as I thought it suppose to work :( https://unix.stackexchange.com/questions/50642/download-all-dependencies-with-yumdownloader-even-if-already-installed does contain some of the info perform this but those all looks cryptic so living it as it is with `--resolve`